### PR TITLE
New voicehints for motorway exit

### PR DESCRIPTION
**Pull request by afischerdev**
This is the PR for #734

The problem with this new voicehints is, not all clients are ready for using.
First rule: the export contains what the clients can read. Means

- Locus already has this tags of Exit Left / Right
- all other become the old values - we can change later

For testing select you favorite turnInstructionMode.
Please use for a direct test export the button between close and export.
When using JSON export you could select `turnInstructionMode = 2` to get the new index.

For OsmAnd exists a special `turnInstructionMode = 8`. This is not reachable via the profile dialog, you have to set this inside the profile editor. It can be used to prepare OsmAnd to the new voicehints before BRouter delivers the new tags on `turnInstructionMode = 3`.

A Test is possible on the test server: [route](https://brouter.de/brouter-test/#map=12/53.3562/10.0729/osm-mapnik-german_style&lonlats=10.085098,53.311964;10.053516,53.390419&profile=car-fast) 

@devemux86 @vshcherb @vodie @menion
Please let us know when you change your rules in your apps, so we can go on to normalize the voicehint situation.

### CODE REVIEW

**Comment by devemux86** [comment]
@afischerdev 

> Please use for a direct test export the button between close and export.

Can you explain what is the difference between the middle "cloud" button and the export button?

**Comment by afischerdev** [comment]
@devemux86 
The export uses the web client internal export, the cloud button exports without reworking.

**Comment by devemux86** [comment]
@afischerdev 
I have seen that GPX navigation tracks can only be exported with the middle "cloud" button.
(and turnInstructionMode 2)

File: brouter-core/src/main/java/btools/router/VoiceHint.java [hunk 1]
```diff
@@ -26,6 +26,8 @@ public class VoiceHint {
   static final int RNLB = 14; // Roundabout left
   static final int TU = 15; // 180 degree u-turn
   static final int BL = 16; // Beeline routing
+  static final int EL = 17; // Beeline routing
+  static final int ER = 18; // Beeline routing
```
Comment (devemux86) [inline brouter-core/src/main/java/btools/router/VoiceHint.java:30]: @afischerdev The comments here should mention the exit instructions.

**Comment by devemux86** [comment]
@afischerdev Thanks for this implementation. :+1: 

Does the test server need a restart?
Routing returns "502 Bad Gateway".

**Comment by afischerdev** [comment]
Just restarted

**Comment by afischerdev** [comment]
Just add a logic that also find motorway exits when there are no more entries.
Therefore I added an new END voicehint type and throw it away after calculation.
This is also already available on testserver.

**Comment by devemux86** [comment]
@afischerdev

Should the clients do something with this?
Or it is handled internally and is mapped to the EL / ER instructions?

**Comment by afischerdev** [comment]
@devemux86 
It was only an internal update to get more exit nodes.
As before the client only has to handle the new voicehints.

**Comment by devemux86** [comment]
@afischerdev 

> As before the client only has to handle the new voicehints.

You mean the 2 new EL + ER instructions.
(not the END)

**Pull request merged**